lib/std/posix.zig

```diff
@@ -7479,7 +7479,7 @@ pub fn ioctl_SIOCGIFINDEX(fd: fd_t, ifr: *ifreq) IoCtl_SIOCGIFINDEX_Error!void {
     }
 }
 
-const lfs64_abi = native_os == .linux and builtin.link_libc and builtin.abi.isGnu();
+const lfs64_abi = native_os == .linux and builtin.link_libc and (builtin.abi.isGnu() or builtin.abi.isAndroid());
```

Comment:
Per your link, these functions are only available from API level 24. Notably, Zig defaults to API level 14:

https://github.com/ziglang/zig/blob/677b2d62e5af4d7a97883994b8d8c45a06535ea0/lib/std/Target.zig#L485

This needs a version check:

```suggestion
const lfs64_abi = native_os == .linux and builtin.link_libc and (builtin.abi.isGnu() or builtin.abi.isAndroid() and builtin.os.version_range.linux.android >= 24);
```

Reply:
Seems quite a few works to be done to work around the older API levels.

https://android-review.googlesource.com/c/platform/bionic/+/429481

Reply:
It seems we basically have no support for Linux with 32-bit file offsets; for glibc, we assume a non-archaic version, and musl has always had 64-bit file offsets. Android would be the first ABI where we'd actually have to support 32-bit file offsets...

... and honestly, I'm not at all convinced that it's worth the trouble. According to [apilevels.com](https://apilevels.com), API level 24 has 97% cumulative usage.

So, you know what, let's undo the version check I suggested here, and instead bump the standard library's minimum Android API level to 24 here:

https://github.com/ziglang/zig/blob/677b2d62e5af4d7a97883994b8d8c45a06535ea0/lib/std/Target.zig#L485

Reply:
> 
> ... and honestly, I'm not at all convinced that it's worth the trouble. According to [apilevels.com](https://apilevels.com), API level 24 has 97% cumulative usage.
> 

Couldn't agree more.